File: Thefieldofdreams.Domain/Entities/Referral.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Thefieldofdreams.Domain.Entities
{
    public class Referral : BaseEntity
    {
        public Guid ReferrerUserId { get; set; }
        public Guid ReferredUserId { get; set; }
        public int PointsAwarded { get; set; }
        public bool IsCompleted { get; set; }
    }
}

File: ThefieldofdreamsClient/src/app/models/referral.ts
export interface Referral {
  id: string;
  referrerUserId: string;
  referredUserId: string;
  pointsAwarded: number;
  isCompleted: boolean;
  createdAt: string;
}

export interface CreateReferralRequest {
  referrerUserId: string;
  referredUserId: string;
  pointsAwarded: number;
}
